sdk/core/Microsoft.Azure.Core.Spatial/src/Serialization/GeographyPointConverter.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Spatial;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Converts between <see cref="GeographyPoint" /> objects and Geo-JSON points.
+    /// </summary>
+    public class GeographyPointConverter : JsonConverter<GeographyPoint>
```

Comment:
What about naming it a `SpatialGeoJSONConverter`. In case we need to add support for more types in the future.

Reply:
(same for newtonsoft)

Reply:
I'd rather have specific ones with a factory method that returns an enumerable of them, which is what track 1 was doing instead. Having a converter do too much not only complicates maintenances, but reduces flexibility. Also, traditionally the name reflects the type being converted + "Converter". This is true of most System.Text.Json and Newtonsoft.Json converters.

Reply:
What if we need to support the base type https://docs.microsoft.com/en-us/dotnet/api/microsoft.spatial.geography?view=odata-spatial-7.0?

I would rather ship one type than 10. That's what we doing for GeoJSON in Azure.Core as well.

Reply:
We should talk about that. I don't think that's a good idea. We could maybe have a container type or a method that just returns an enumerable, but you don't typically see that level of granularity (or lack there of) with other converters. Why not allow flexibility to pick and choose which types should be supported?

Reply:
I wonder who needs that level of flexibility? I'm not sure there is a customer scenario where being able to strictly deserialize some GeoJSON types but not the others is more desirable than just having turnkey support for all of them.

Reply:
Why don't other libraries' converters be more general / over-arching? Choice. Besides, if we do make more general ones, which types do they all support? It wouldn't be clear to callers. If this just handles `Geography` (in which case I'd still want to call it `GeographyConverter`) and a polygon fails to deserialize in at least some versions of the library, how would a caller know?

Reply:
>Why don't other libraries' converters be more general / over-arching?

Different libraries have different incentives/motivation. A major thing we are trying to do in Track 2 libraries is to ship fewer types unless we have a reason to. It's easy to add separate types if anyone asks for them but impossible to take them away.

>a polygon fails to deserialize in at least some versions of the library.

Why would it fail?

Reply:
Because I don't need to write a converter for it right now. The current target is Search, which only needs to deserialize a point. Someday when we add support for polygon and/or lines, collections, etc. it will work. But then if someone uses an older version it won't. Or a newer version may produce unexpected results when new types are supported. I don't see how that's a good experience. In Azure SDK, we also try to avoid breaking behaviors, not just breaking code changes.

We could, instead, create a container type that wraps these (`CanConvert` could be made more efficient in that case for all supported types), and people can use that if they don't care about unexpected behavior changes.

Reply:
Yeah, I'm fine if we start by not supporting.

How is going from throwing an exception for unsupported types to not throwing is breaking?

Can you give an example of unexpected behavior change?

Reply:
I'm not throwing right now since I'm only declaring support for `GeographyPoint`, i.e. `JsonConverter<GeographyPoint>`. So if someone deserializes something with a different object, it will be treated as a complex object and deserialized differently than if it were suddenly supported by a converter they had already added, e.g. a `GeographyConverter`. Now they get back a different type from the converter, which will likely cause their deserialization to fail (e.g. failure to cast).

A single converter than handles a growing list of types will change behavior. It's not like adding a method or model that, if you don't use it, shouldn't matter. This is a type that everyone who needs it would've added, and the behavior would otherwise change on them when the package is upgraded. That's why I'm suggesting, if we want a single type people can add if they don't worry about behavior changes, we can either add a container class or factory method, which track 1 uses.

@KrzysztofCwalina or @tg-msft what are your thoughts?

Reply:
So how do you envision this library looking when we support all the geometry types?

Reply:
As discussed, I'll name this generically for now. I opened #15506 to track. I could probably get that done soon, but not right now. I will rename this type generically.